### PR TITLE
Enable closing of fuzzyfinder from the caller by passing context

### PR DESCRIPTION
**Pull request by ahsan**
This PR adds a new option called `WithContext`. This enables callers of `fuzzyfinder.Find` to cancel the passed context and  forcefully make `Find` return if completing it is no longer desirable.

An example use case is when `Find` is called with the `WithHotReload` option, but the caller does not have any items to put in the slice (e.g due to a failed network call).

Please let me know what you think!

### CODE REVIEW

File: fuzzyfinder.go [hunk 1]
```diff
@@ -94,6 +98,12 @@ func (f *finder) initFinder(items []string, matched []matching.Matched, opt opt)
 		if err := f.term.Init(); err != nil {
 			return errors.Wrap(err, "failed to initialize screen")
 		}
+
+		eventsChan := make(chan tcell.Event)
+		quitChan := make(chan struct{})
```
Comment (ktr0731) [inline fuzzyfinder.go:103]: This channel is not used anywhere so we can use a nil channel.

File: fuzzyfinder.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ import (
 var (
 	// ErrAbort is returned from Find* functions if there are no selections.
 	ErrAbort   = errors.New("abort")
+	ErrCancel  = errors.New("cancel")
```
Comment (ktr0731) [inline fuzzyfinder.go:28]: I think that we don't need a self-defined cancellation error. Context is a standard mechanism, so how about returning` ctx.Err()` simply?

**Review by ktr0731 — COMMENTED** [review]
Thanks for your contribution!
I think that it makes sense to provide an option that accepts context 👍 
Almost the implementation looks good, but I suggested some trivial points. Could you work on them when you have time?

File: fuzzyfinder.go [hunk 3]
```diff
@@ -442,9 +448,10 @@ func (f *finder) draw(d time.Duration) {
 }
 
 // readKey reads a key input.
-// It returns ErrAbort if esc, CTRL-C or CTRL-D keys are inputted.
-// Also, it returns errEntered if enter key is inputted.
-func (f *finder) readKey() error {
+// It returns ErrAbort if esc, CTRL-C or CTRL-D keys are inputted,
+// errEntered in case of enter key, and ErrCancel when the passed
```
Comment (ktr0731) [inline fuzzyfinder.go:452]:
```suggestion
// errEntered in case of enter key, and a context error when the passed
```

Comment (ahsan) [reply fuzzyfinder.go]: Ahh I missed this.. thanks!

**Review by ktr0731 — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**